### PR TITLE
Add upgrade notes for 'set_external_tags'

### PR DESCRIPTION
**Pull request by masci**
### What does this PR do?

Add upgrade notes for a change in one of the functions of the internal Python API exposed by the Agent.

### Motivation

Function behaves differently now, custom checks should be updated accordingly.

### CODE REVIEW

**Review by olivielpeau — APPROVED** [review]
✨ 

Note (nit): if we need to add other upgrade notes about the python modules, we should probably group them together for clarity. A simple way to do that would be to group them all in this release note file.

**Review by apigirl — APPROVED** [review]

File: releasenotes/notes/change-set-external-tags-api-e5d6f80ad7c8376f.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    A change in the 'set_external_tags' function from the 'datadog_agent' Python
```
Comment (hush-hush) [inline releasenotes/notes/change-set-external-tags-api-e5d6f80ad7c8376f.yaml]: We still drop invalid tags silently but with #3562 we handle "ascii unicode" like before. I think we should still keep this note as all direct call to Go functions from Python should go through the base class.

**Comment by olivielpeau** [comment]
@masci let's merge?

**Comment by masci** [comment]
I lean towards merging but notice https://github.com/DataDog/datadog-agent/pull/3562 is mutually exclusive

**Comment by masci** [comment]
Closing in favor of https://github.com/DataDog/datadog-agent/pull/3562

**Pull request closed without merging**